wordpress-mobile/WordPress-Android
# Jetpack Mobile: Scan menu

Parent #13326

It displays scan menu on My Site for sites with scan capability for a user.

**To test**

Prerequisite:  Make sure both Scan feature flag and MySiteImprovements flag are set to on in the App settings.
(Changes applied only to MySiteImprovements flow)

Test that scan menu is shown on My Site only when the user has scan feature unlocked.

|My Site Scan Menu|
|--------------------|
|![screnshot](https://user-images.githubusercontent.com/1405144/101352414-1e64ea00-38b8-11eb-9fc2-d6b6c2feeeee.png)|


Case | Site Type | Login Type | Sub Type  |  User Role | Show Scan |  Scan State API Response (Reason)
-- | -- | -- | -- | -- | -- | --
1a | **.com** |   | simple |  | ❌ | unavailable (wpcom_site)
1b |   |   | atomic | admin | ✅ | available - idle or scanning
1c |   |   |   | non admin | ❌  |  error (authorization_required)
2a | **Jetpack** | wp.com login | plan without scan feature |  | ❌ | unavailable (missing_scan_capability)
2b |   |   | plan with scan feature | admin | ✅ | available - idle or scanning
2c |   |   |   | non admin | ❌  |  error (authorization_required)
2d |   | self-hosted login | plan without scan feature  |  | ❌ |  error (authorization_required)
2e |   |   | plan with scan feature |   | ❌ |  error (authorization_required)

**Questions**

2d, 2e - Scan state rest API gives `authorization_required` error for self-hosted login for a Jetpack site. Should we always display scan menu via this flow? - Answered in https://github.com/wordpress-mobile/WordPress-Android/pull/13555#issuecomment-740625764

Not in scope
Scan menu icon fine-tuning
Click action

PR submission checklist:
